sku.0/sys.server/compiled/game/script/item/special/no_trade_removable.java

```diff
@@ -0,0 +1,46 @@
+package script.item.special;
+
+import script.obj_id;
+import script.string_id;
+import script.menu_info;
+import script.menu_info_types;
+
+/**
+ * Allows items that previously had the "noTrade" flag to be made tradeable by
+ * their original owner.
+ * 
+ * Automatically attached to objects in no_trade_removable.tab
+ */
+public class no_trade_removable extends script.base_script {
+    public no_trade_removable() {
+    }
+
+    // Using one of the unused reserved entries.
+    public static final int RADIAL_MENU_ENTRY = menu_info_types.SERVER_MENU47;
+    public static final string_id REMOVE_NO_TRADE = new string_id("ui_radial", "no_trade_removable");
+
+    public int OnObjectMenuRequest(obj_id self, obj_id player, menu_info mi) throws InterruptedException {
+        if (hasObjVar(self, "noTrade") && (getOwner(self) == player || isGod(player))) {
```

Comment:
You should also validate that the object is contained by the owner's inventory here. There isn't any need to populate this context menu option on objects that are elsewhere (and because hackers).